backend/src/org/commcare/suite/model/StackFrameStep.java

```diff
@@ -86,19 +87,17 @@ public boolean getValueIsXPath() {
         return valueIsXpath;
     }
 
-    public void addExtra(String key, String value) {
+    /**
+     * @param value Must extend Externalizable or be a basic data type (String, Vector, etc)
```

Comment:
`public <T extends Externalizable> void addExtra(String key, T value)` would take care of the Externalizable portion, but I don't know of an encompassing class for "basic data types"